### PR TITLE
DOC: update release notes with all deprecations for 1.12 release

### PR DESCRIPTION
**Pull request by j-bowhay**
#### What does this implement/fix?

Adds a couple missing entries to the deprecation section of the release notes

#### Additional information

cc @h-vetinari @tylerjereddy

### CODE REVIEW

File: doc/source/release/1.12.0-notes.rst [hunk 1]
```diff
@@ -273,6 +273,14 @@ Deprecated features
 - All wavelet functions have been deprecated, as PyWavelets provides suitable
   implementations; affected functions are: ``signal.{daub, qmf, cascade,
   morlet, morlet2, ricker, cwt}``
+- `scipy.integrate.trapz`, `scipy.integrate.cumtrapz`, and `scipy.integrate.simps` have
+  been deprecated in favour of `scipy.integrate.trapezoid`, `scipy.integrate.cumulative_trapezoid`,
+  and `scipy.integrate.simpson` and will be removed in SciPy 1.14.
```
Comment (j-bowhay) [inline doc/source/release/1.12.0-notes.rst:278]:
```suggestion
  and `scipy.integrate.simpson` respectively and will be removed in SciPy 1.14.
```

File: doc/source/release/1.12.0-notes.rst [hunk 2]
```diff
@@ -281,8 +289,11 @@ Expired Deprecations
 There is an ongoing effort to follow through on long-standing deprecations.
 The following previously deprecated features are affected:
 
-- The ``centered`` keyword of `stats.qmc.LatinHypercube` has been removed.
+- The ``centered`` keyword of `scipy.stats.qmc.LatinHypercube` has been removed.
   Use ``scrambled=False`` instead of ``centered=True``.
+- ``scipy.stats.binom_test`` has been removed in favour of `scipy.stats.binomtest`.
+- The use of ``scale='raw'`` has been removed in favour of ``'scale=1'`` in
+  `scipy.stats.iqr`.
```
Comment (j-bowhay) [inline doc/source/release/1.12.0-notes.rst:296]:
```suggestion
- In `scipy.stats.iqr`, the use of ``scale='raw'`` has been removed in favour 
  of ``'scale=1'``.
```

File: doc/source/release/1.12.0-notes.rst [hunk 3]
```diff
@@ -281,8 +289,11 @@ Expired Deprecations
 There is an ongoing effort to follow through on long-standing deprecations.
 The following previously deprecated features are affected:
 
-- The ``centered`` keyword of `stats.qmc.LatinHypercube` has been removed.
+- The ``centered`` keyword of `scipy.stats.qmc.LatinHypercube` has been removed.
   Use ``scrambled=False`` instead of ``centered=True``.
+- ``scipy.stats.binom_test`` has been removed in favour of `scipy.stats.binomtest`.
+- The use of ``scale='raw'`` has been removed in favour of ``'scale=1'`` in
```
Comment (j-bowhay) [inline doc/source/release/1.12.0-notes.rst:295]:
```suggestion
- The use of ``scale='raw'`` has been removed in favour of ``scale=1`` in
```

File: doc/source/release/1.12.0-notes.rst [hunk 4]
```diff
@@ -273,6 +273,14 @@ Deprecated features
 - All wavelet functions have been deprecated, as PyWavelets provides suitable
   implementations; affected functions are: ``signal.{daub, qmf, cascade,
   morlet, morlet2, ricker, cwt}``
+- `scipy.integrate.trapz`, `scipy.integrate.cumtrapz`, and `scipy.integrate.simps` have
```
Comment (j-bowhay) [inline doc/source/release/1.12.0-notes.rst:276]:
```suggestion
- ``scipy.integrate.trapz``, ``scipy.integrate.cumtrapz``, and ``scipy.integrate.simps`` have
```

**Review by lucascolley — APPROVED** [review]

File: doc/source/release/1.12.0-notes.rst [hunk 5]
```diff
@@ -273,6 +273,14 @@ Deprecated features
 - All wavelet functions have been deprecated, as PyWavelets provides suitable
   implementations; affected functions are: ``signal.{daub, qmf, cascade,
   morlet, morlet2, ricker, cwt}``
+- ``scipy.integrate.trapz``, ``scipy.integrate.cumtrapz``, and ``scipy.integrate.simps`` have
+  been deprecated in favour of `scipy.integrate.trapezoid`, `scipy.integrate.cumulative_trapezoid`,
+  and `scipy.integrate.simpson` respectively and will be removed in SciPy 1.14.
+- The deprecation of the ``tol`` argument of ``scipy.sparse.linalg.{gcrotmk,lgmres,minres,tfqmr}``
+  is now consistent with the other iterative solvers and will be removed in SciPy 1.14
+  in favour of ``rtol``. Further, the default value of ``atol`` for these functions is due
+  to change to ``0.0`` in SciPy 1.14.
```
Comment (h-vetinari) [inline doc/source/release/1.12.0-notes.rst:282]: The deprecation of `tol` happened across the board for all* sparse solvers in 1.12, see #18488.

\* Of course, there's an [exception](https://github.com/scipy/scipy/issues/19704)... 😑

**Review by h-vetinari — COMMENTED** [review]
The note about the sparse solvers is not correct/complete.

File: doc/source/release/1.12.0-notes.rst [hunk 6]
```diff
@@ -273,6 +273,14 @@ Deprecated features
 - All wavelet functions have been deprecated, as PyWavelets provides suitable
   implementations; affected functions are: ``signal.{daub, qmf, cascade,
   morlet, morlet2, ricker, cwt}``
+- ``scipy.integrate.trapz``, ``scipy.integrate.cumtrapz``, and ``scipy.integrate.simps`` have
+  been deprecated in favour of `scipy.integrate.trapezoid`, `scipy.integrate.cumulative_trapezoid`,
+  and `scipy.integrate.simpson` respectively and will be removed in SciPy 1.14.
+- The deprecation of the ``tol`` argument of ``scipy.sparse.linalg.{gcrotmk,lgmres,minres,tfqmr}``
+  is now consistent with the other iterative solvers and will be removed in SciPy 1.14
```
Comment (j-bowhay) [inline doc/source/release/1.12.0-notes.rst:280]: I believe this is now correct @h-vetinari 
```suggestion
- The ``tol`` argument of ``scipy.sparse.linalg.{gcrotmk,lgmres,minres,tfqmr,bcg,bicstab,cg,cgs,gmres,qmr}``
  is now deprecated and will be removed in SciPy 1.14
```

Comment (h-vetinari) [reply doc/source/release/1.12.0-notes.rst:280]: Let's pull forward the mention of `rtol` and alphabetize the solvers
```suggestion
- The ``tol`` argument of ``scipy.sparse.linalg.{bcg,bicstab,cg,cgs,gcrotmk,gmres,lgmres,minres,qmr,tfqmr}``
  is now deprecated in favour of ``rtol`` and will be removed in SciPy 1.14.
```

**Review by h-vetinari — APPROVED** [review]
Modulo a slight refomulation this LGTM, so I'll put this in. Thanks! :)

File: doc/source/release/1.12.0-notes.rst [hunk 7]
```diff
@@ -273,6 +273,14 @@ Deprecated features
 - All wavelet functions have been deprecated, as PyWavelets provides suitable
   implementations; affected functions are: ``signal.{daub, qmf, cascade,
   morlet, morlet2, ricker, cwt}``
+- ``scipy.integrate.trapz``, ``scipy.integrate.cumtrapz``, and ``scipy.integrate.simps`` have
+  been deprecated in favour of `scipy.integrate.trapezoid`, `scipy.integrate.cumulative_trapezoid`,
+  and `scipy.integrate.simpson` respectively and will be removed in SciPy 1.14.
+- The ``tol`` argument of ``scipy.sparse.linalg.{bcg,bicstab,cg,cgs,gcrotmk,gmres,lgmres,minres,qmr,tfqmr}``
+  is now deprecated in favour of ``rtol`` and will be removed in SciPy 1.14.
+  in favour of ``rtol``. Further, the default value of ``atol`` for these functions is due
+  to change to ``0.0`` in SciPy 1.14.
```
Comment (h-vetinari) [inline doc/source/release/1.12.0-notes.rst:282]:
```suggestion
- The ``tol`` argument of ``scipy.sparse.linalg.{bcg,bicstab,cg,cgs,gcrotmk,gmres,lgmres,minres,qmr,tfqmr}``
  is now deprecated in favour of ``rtol`` and will be removed in SciPy 1.14.
  Furthermore, the default value of ``atol`` for these functions is due
  to change to ``0.0`` in SciPy 1.14.
```

**Comment by h-vetinari** [comment]
Doc build is broken by https://github.com/scipy/scipy/issues/19793. Since we're going to have to fix that separately anyway, I'm merging here.

**Pull request merged**